ddtrace/span.py

```diff
@@ -498,12 +498,13 @@ def pprint(self):
 
     @property
     def context(self):
-        """
-        Property that provides access to the ``Context`` associated with this ``Span``.
-        The ``Context`` contains state that propagates from span to span in a
-        larger trace.
-        """
-        return self._context
+        # type: () -> Context
+        """Return the trace context for this span."""
+        if self._context is None:
+            ctx = Context(trace_id=self.trace_id, span_id=self.span_id)
```

Comment:
Do we need to set `self._context` here if we have to create a new context?

Reply:
Yeah the semantics here are a little weird. Any time a span is created with start_span the context is provided to the span so `self._context` will always have a value.

So the only case in which this branch will be taken is if the span is created manually without a context. All that said though, I think it does make sense to set `self._context` to the new context created because otherwise the following case would fail:

```python
span1 = Span(None, "span1")
span2 = Span(None, "span2", context=span1.context)
span2.context.dd_origin = "test"
assert span1.context.dd_origin == "test"  # fails!
```

Definitely not a supported use-case but I think we can fix this case just so it makes sense

Reply:
Addressed in 2a435cd